keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlVerificationRepository.kt

```diff
@@ -100,19 +101,20 @@ class SqlVerificationRepository(
               .set(ENVIRONMENT_LAST_VERIFIED.ARTIFACT_VERSION, artifactVersion)
               .set(ENVIRONMENT_LAST_VERIFIED.AT, now)
               .onDuplicateKeyUpdate()
+              .set(ENVIRONMENT_LAST_VERIFIED.ARTIFACT_VERSION, artifactVersion)
               .set(ENVIRONMENT_LAST_VERIFIED.AT, now)
               .execute()
           }
-          .map { (_, deliveryConfigName, _, environmentName, _, artifactReference, artifactVersion) ->
-            VerificationContext(
-              deliveryConfigByName(deliveryConfigName),
-              environmentName,
-              artifactReference,
-              artifactVersion
-            )
-          }
       }
     }
+      .map { (_, deliveryConfigName, _, environmentName, _, artifactReference, artifactVersion) ->
+        VerificationContext(
+          deliveryConfigByName(deliveryConfigName),
+          environmentName,
+          artifactReference,
+          artifactVersion
+        )
+      }
```

Comment:
Moved this outside of the transaction block so if it throws an exception it doesn't cause the transaction to roll back.

Reply:
Good catch!

Reply:
Trying to figure out a good way to test it